lib/msal-browser/docs/errors.md

```diff
@@ -114,8 +114,85 @@ msalInstance.handleRedirectPromise()
-- [msal-angular errors](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/v2-docs/errors.md)
+- [msal-react errors](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-react/docs/errors.md#interaction_in_progress)
+- [msal-angular errors](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/v2-docs/errors.md#interaction_in_progress)
+
+If you are not using any of the wrapper libraries but concerned that your application might trigger concurrent interactive requests, you should check if any other interaction is in progress prior to invoking an interaction in your token acquisition method. You can achieve this by implementing a global application state or a broadcast service etc. that emits the current MSAL interaction status via [MSAL Events API](./events.md).
+
+❌ The following example may throw this error because the `acquireTokenPopup` in the **catch** block does not check if there is another interaction taking place at the moment:
+
+```javascript
+async function myAcquireToken(request) {
+    const msalInstance = getMsalInstance(); // get the msal application instance
+
+    const tokenRequest = {
+        account: msalInstance.getActiveAccount() || null;
+        ...request
+    };
+
+    let tokenResponse;
+
+    try {
+        // attempt silent acquisition first
+        tokenResponse = await msalInstance.acquireTokenSilent(request);
+    } catch (error) {
+        if (error instanceof InteractionRequiredAuthError) {
+            try {
+                tokenResponse = await msalInstance.acquireTokenPopup(request);
+            } catch (err) {
+                console.log(err);
+                // handle other errors
+            }
+        }
+
+        console.log(error);
+        // handle other errors
+    }
+
+    return tokenResponse;
+}
+```
+
+✔️ To resolve, you should wait for the interaction status to be `None` before calling any other interactive API:
+
+```javascript
+async function myAcquireToken(request) {
+    const msalInstance = getMsalInstance(); // get the msal application instance
+
+    const tokenRequest = {
+        account: msalInstance.getActiveAccount() || null;
+        ...request
+    };
+
+    let tokenResponse;
+
+    try {
+        // attempt silent acquisition first
+        tokenResponse = await msalInstance.acquireTokenSilent(request);
+    } catch (error) {
+        if (error instanceof InteractionRequiredAuthError) {
+            try {
+                /**
+                 * "myWaitFor" method polls the interaction status via getInteractionStatus() from
+                 * the application state and resolves when it's equal to "None".
+                 */
+                await myWaitFor(() => myGlobalState.getInteractionStatus() === InteractionStatus.None);
```

Comment:
Rather than delaying execution of acquireTokenPopup, resulting in potentially multiple popups back to back, if interaction is in progress currently, we should wait and then go back to the top (to call ATS again which should now succeed in theory) else if interaction is not already in progress call popup.

Reply:
ah, good idea! will revise

Reply:
revised -let me know what you think